trulens_eval/trulens_eval/feedback/provider/bedrock.py

```diff
@@ -0,0 +1,1028 @@
+import logging
+import os
+
+from trulens_eval.feedback import prompts
+from trulens_eval.feedback.provider.base import Provider
+from trulens_eval.utils.generated import re_1_10_rating
+
+import json
+
+logger = logging.getLogger(__name__)
+
+
+class Bedrock(Provider):
```

Comment:
Instead of dup code; can we do inheritance? 

Parent provider class something like LLMProvider(Prodvider) (has all the dup code)

and do 
```
OpenAI(LLMProvider):
    @override
    def _create_chat_completion
    @override
    etd
```
and 

```
Bedrock(LLMProvider):
    @override
    def _create_chat_completion
    @override
    etd
```

Reply:
The usage examples are provider specific so we'd need to sacrifice that to inherit the methods with the provider implementation abstracted

Reply:
The usage examples can be unique; you just add the specific one in the override and then call the super() method. Code duplication is very hard to maintain

Reply:
Also; the usage examples are barely provider specific; right? At this point its probably better to just have mkdocs show the base class LLMProvider and have the docs there show two pieces. 

for OpenAI
```
import trulens....OpenAI
provider = OpenAI()
```

for Bedrock
```
import trulens....Bedrock
provider = Bedrock()
```

```
rest of the usage is the same
```

Reply:
Also; if model_engine is an issue; just make model_engine do nothing in bedrock and move it to the parent class and note that it is currently only needed for OpenAI() provider

Reply:
Can you take another look @rshih32 ? I factored out all of the LLM feedbacks into base and just overriding where necessary in openai, litellm, bedrock